apps/frontend/core-dapp/src/features/trade/open-trade/OpenTradeButton.tsx

```diff
@@ -0,0 +1,65 @@
+import { observer } from 'mobx-react-lite'
+import { Button } from 'prepo-ui'
+import { useMemo } from 'react'
+import { useRootStore } from '../../../context/RootStoreProvider'
+import ConnectButton from '../../connect/ConnectButton'
+
+const OpenTradeButton: React.FC = () => {
+  const { preCTTokenStore, tradeStore, web3Store } = useRootStore()
+  const { balanceOfSigner } = preCTTokenStore
+  const { connected, isNetworkSupported } = web3Store
+  const {
+    insufficientBalanceForOpenTrade,
+    needApproval,
+    selectedMarket,
+    openTradeAmountBN,
+    openTradeButtonLoading,
+    withinBounds,
+    tradeDisabled,
+  } = tradeStore
+
+  const buttonText = useMemo(() => {
+    if (!selectedMarket) return 'Select a Market'
+
+    if (withinBounds === false) return 'Unprofitable Trade'
+
+    if (connected) {
+      if (insufficientBalanceForOpenTrade) return 'Insufficient Balance'
+      if (balanceOfSigner?.gt(0) && needApproval) return 'Approve'
+    }
+
+    return 'Trade'
+  }, [
+    balanceOfSigner,
+    connected,
+    insufficientBalanceForOpenTrade,
+    needApproval,
+    selectedMarket,
+    withinBounds,
+  ])
+
+  const overrideConnectButton = withinBounds === false
+
+  if ((!connected || !isNetworkSupported) && !overrideConnectButton) return <ConnectButton />
+
+  const handleClick = (): void => {
+    if (needApproval) {
+      tradeStore.approve()
+    } else {
+      tradeStore.openTrade()
+    }
+  }
+
+  return (
+    <Button
+      block
+      disabled={tradeDisabled || !selectedMarket || openTradeAmountBN?.eq(0) || !withinBounds}
```

Comment:
```suggestion
      disabled={!selectedMarket || openTradeAmountBN?.eq(0) || !withinBounds}
```

remove tradeDisabled